identity-and-access-management/key-manager/api-cli/manage-keys-with-tink.mdx

```diff
@@ -0,0 +1,266 @@
+
+    secret1, _ := primitive.Encrypt([]byte("This is a secret"), dataAssociatedData1)
+    /*
+    *   Store secret1 somewhere
+    */
+
+    secret2, _ := primitive.Encrypt([]byte("This is another secret"), dataAssociatedData2)
+    /*
+    *   Store secret2 somewhere
+    */
+    ```
+    <Message type="note">
+     Tink only provides methods that work on types that comply with `Reader` and `Writer` interfaces. If you need to write and read Tink keysets as direct bytes, you can use `bytes.Buffer`, which is an in-memory buffer used to hold the encrypted keyset. This allows you to serialize and deserialize keysets directly as byte arrays instead of using files.
+    </Message>
+
+3. Run the following command to store the DEK in memory as bytes:
+
+    ```go
+    buf := new(bytes.Buffer)
+    w := keyset.NewBinaryWriter(buf)
+    handle.WriteWithAssociatedData(w, kekAEAD, associatedData)
+    encDEK := buf.Bytes()  // encrypted DEK in Tink wire format
+    ```
+
+You can then store the bytes of the encrypted DEK in a database for example, with the encrypted data it protects. For example, the encrypted data (enc_data) and the encrypted DEK (enc_dek) might be stored together in a row in a database (base64-encoded in the following example):
+
+    ```console
+    SELECT id, enc_data, enc_dek FROM sensible_stuff;
+
+    id  | enc_data                   | enc_dek
+    -----|----------------------------|-------------------
+    42  | "7NXIqRms0+TiKj+V0gv1s..." | "vIiYBeypb7Yk..."
+    43  | "7X8v0GVrXWwL/ckzfRms0..." | "vIiYBeypb7Yk..."
+    ...
+    ...
+    ...
+    ```
+
+
+## Associated Data
+
+Associated Data (AD) is not encrypted, but it is authenticated. It must be the same when you encrypt and decrypt data, otherwise the decryption fails. This is useful to prevent reading the wrong data in the wrong context. In the table above, the data in both rows 42 and 43 is protected by the same DEK. If we swapped the data, an application would be able to decrypt the data from another row. But, by providing the intended ID as the associated data, the decryption would fail.
+
+### Encrypt data with AD before inserting it
+
+Run the following command to encrypt your data with `Associated Data`. In the example below, associated data like `id42` and `id43` is used to ensure that data from row 42 cannot be decrypted in the context of row 43.
+
+    ```go
+    handle, _ := keyset.ReadWithAssociatedData(r, kekAEAD, dekAD)
+    primitive, _ := aead.New(handle)  // Same DEK for the two payloads
+
+    secret1, _ := primitive.Encrypt([]byte("This is a secret"), []byte("id42"))
+    // Insert secret1 into row 42
+
+    secret2, _ := primitive.Encrypt([]byte("This is another secret"), []byte("id43"))
+    // Insert secret2 into row 43
+    ```
+
+Associated Data does not need to be stored, as it can be infered from the context at decryption time. It is also possible to use a unique DEK for each payload. We recommend using Associated Data.
```

Comment:
```suggestion
Associated Data does not need to be stored, as it can be inferred from the context at decryption time. It is also possible to use a unique DEK for each payload. We recommend using Associated Data.
```